plugins/woocommerce/src/Blocks/BlockTypes/AddToCartForm.php

```diff
@@ -136,7 +136,13 @@ protected function render( $attributes, $content, $block ) {
 		}
 
 		$is_external_product_with_url = $product instanceof \WC_Product_External && $product->get_product_url();
-		$is_stepper_style             = 'stepper' === $attributes['quantitySelectorStyle'] && ! $product->is_sold_individually() && Features::is_enabled( 'add-to-cart-with-options-stepper-layout' );
+		$managing_stock               = $product->managing_stock();
+		$stock_quantity               = $product->get_stock_quantity();
+
+		/**
+		 * The stepper buttons don't show when the product is sold individually or stock quantity is less or equal to 1 because the quantity input field is hidden.
+		 */
+		$is_stepper_style = 'stepper' === $attributes['quantitySelectorStyle'] && ! $product->is_sold_individually() && ( ( $managing_stock && $stock_quantity > 1 ) || ! $managing_stock ) && Features::is_enabled( 'add-to-cart-with-options-stepper-layout' );
```

Comment:
Ups, sorry I missed in the first review, but in the _Add to Cart with Options_ block there will only be the stepper style (https://github.com/woocommerce/woocommerce/issues/56686), so this variable will be removed. I think it would be easier to simply tweak the _Quantity Selector_ block and [return early](https://github.com/woocommerce/woocommerce/blob/7edde48eaf24e719ade0c1fcb1446d0aaf06c019/plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptionsQuantitySelector.php#L135) when there is only one item in stock?